dhall-docs/src/Dhall/Docs/Core.hs

```diff
@@ -323,8 +323,8 @@ createIndexes packageName files = map toIndex dirToDirsAndFilesMapAssocs
         initialMap :: Map (Path Rel Dir) [Path Rel Dir]
         initialMap = Map.fromList $ map (,[]) dirs
 
-        go :: Map (Path Rel Dir) [Path Rel Dir] -> Path Rel Dir -> Map (Path Rel Dir) [Path Rel Dir]
-        go dirMap d = Map.adjust ([d] <>) (key $ Path.parent d) dirMap
+        go :: Path Rel Dir -> Map (Path Rel Dir) [Path Rel Dir] -> Map (Path Rel Dir) [Path Rel Dir]
+        go d dirMap = Map.adjust ([d] <>) (key $ Path.parent d) dirMap
```

Comment:
IAnother solution would be to switch the adjust function here:

```suggestion
        go d dirMap = Map.adjust (<> [d]) (key $ Path.parent d) dirMap
```

but better stick with `foldr`

Reply:
Guilty as charged.  ;-)

Your call on which solution you prefer, but `foldr` is maybe more "canonical" to my mind for keeping the order.

Reply:
Let's stick with `foldr`. In almost all cases it performs better than `foldl`.